Intermittent UI test failure - < SettingsAboutTest. verifyAboutFirefoxPreview >

### Firebase Test Run: [Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6236068061031912357/executions/bs.832797858bf53afb/testcases/3/test-cases)
### Stacktrace:
androidx.test.espresso.PerformException: Error performing 'single click' on view '(view.getId() is <2131297114/org.mozilla.fenix.debug:id/menuButton>)'.
	at androidx.test.espresso.PerformException$Builder.build(PerformException.java:1)
	at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:9)
	at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:8)
	at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:11)
	at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:4)
	at org.mozilla.fenix.ui.robots.HomeScreenRobot$Transition.openThreeDotMenu(HomeScreenRobot.kt:370)
	at org.mozilla.fenix.ui.SettingsAboutTest.verifyAboutFirefoxPreview(SettingsAboutTest.kt:93)
	... 34 trimmed
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() covers at least <90> percent of the view's area)
Target view: "AppCompatImageView{id=2131297114, res-name=menuButton, visibility=VISIBLE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@82b6ffe, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"
	at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:20)
	at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:1)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:2)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
    
### Build: 10/13 Main

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21792)

Flaky 1x on 10/26 Main [Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5579494643146576782/executions/bs.51611085ec969cb7/testcases/4/test-cases)

Flaky 1x on 10/28 Main [Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5717243998479719829/executions/bs.4d585c2ee82e2c46/testcases/4/test-cases)

androidx.test.espresso.PerformException: Error performing 'single click' on view '(view.getId() is <2131297114/org.mozilla.fenix.debug:id/menuButton>)'.
	at androidx.test.espresso.PerformException$Builder.build(PerformException.java:1)
	at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:9)
	at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:8)
	at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:11)
	at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:4)
	at org.mozilla.fenix.ui.robots.HomeScreenRobot$Transition.openThreeDotMenu(HomeScreenRobot.kt:380)
	at org.mozilla.fenix.ui.SettingsAboutTest.verifyAboutFirefoxPreview(SettingsAboutTest.kt:93)
	... 34 trimmed
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() covers at least <90> percent of the view's area)
Target view: "AppCompatImageView{id=2131297114, res-name=menuButton, visibility=VISIBLE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@80f4c4b, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"
	at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:20)
	at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:1)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:2)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

Flaky 1x in 11/3 Main [Fireabase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7056909787559757833/executions/bs.10497927c3e9c9a1/testcases/4/test-cases)

androidx.test.espresso.PerformException: Error performing 'single click' on view '(view.getId() is <2131297114/org.mozilla.fenix.debug:id/menuButton>)'.
	at androidx.test.espresso.PerformException$Builder.build(PerformException.java:1)
	at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:9)
	at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:8)
	at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:11)
	at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:4)
	at org.mozilla.fenix.ui.robots.HomeScreenRobot$Transition.openThreeDotMenu(HomeScreenRobot.kt:377)
	at org.mozilla.fenix.ui.SettingsAboutTest.verifyAboutFirefoxPreview(SettingsAboutTest.kt:93)
	... 34 trimmed
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() covers at least <90> percent of the view's area)
Target view: "AppCompatImageView{id=2131297114, res-name=menuButton, visibility=VISIBLE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@8d0f8f7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"
	at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:20)
	at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:1)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:2)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

--- a/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt
+++ b/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt
@@ -44,6 +44,7 @@ import org.junit.Assert.assertTrue
 import org.mozilla.fenix.R
 import org.mozilla.fenix.helpers.Constants.LISTS_MAXSWIPES
 import org.mozilla.fenix.helpers.Constants.PackageName.GOOGLE_PLAY_SERVICES
+import org.mozilla.fenix.helpers.Constants.RETRY_COUNT
 import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
 import org.mozilla.fenix.helpers.TestHelper.appName
 import org.mozilla.fenix.helpers.TestHelper.getStringResource
@@ -575,7 +576,21 @@ private fun rateOnGooglePlayHeading(): UiObject {
 }
 
 private fun aboutFirefoxHeading(): UiObject {
-    settingsList().scrollToEnd(LISTS_MAXSWIPES)
+    for (i in 1..RETRY_COUNT) {
+        try {
+            settingsList().scrollToEnd(LISTS_MAXSWIPES)
+            assertTrue(
+                mDevice.findObject(UiSelector().text("About $appName"))
+                    .waitForExists(waitingTime),
+            )
+
+            break
+        } catch (e: AssertionError) {
+            if (i == RETRY_COUNT) {
+                throw e
+            }
+        }
+    }
     return mDevice.findObject(UiSelector().text("About $appName"))
 }
 

